lib/MemoryModel/MemModel.cpp

```diff
@@ -189,7 +189,7 @@ bool SymbolTableInfo::computeGepOffset(const llvm::User *V, LocationSet& ls) {
         // Handling array types, skipe array handling here
         // We treat whole array as one, then we can distinguish different field of an array of struct
         // e.g. s[1].f1 is differet from s[0].f2
-        if(isa<ArrayType>(*gi))
+        if(isa<ArrayType>(gi.getIndexedType()))
```

Comment:
Alright, sat down and I think I have a strategy to bridge this.  I just want to try this on a few small scale experiments considering the discussion @dtzWill pointed out as hopefully this bridge will provide functionality but can be removed with some ease at a later date...